src/Checkbox/CheckboxComponent.jsx

```diff
@@ -13,10 +12,12 @@ export default observer(() => {
         id="example-checkbox"
         name="foo"
         onChange={(event) => {
-          checkboxPresenter.setIsChecked(event.target.checked)
+          dispatch(
+            {type: 'SET_IS_CHECKED', payload: event.target.checked}
+          );
         }}
-        checked={checkboxPresenter.isChecked}
+        checked={state.isChecked}
```

Comment:
No need to set this to `@observable`.

It's really simple, just a property of the global `state`.